### ISSUE TITLE
Create a mock MongoDB

### ISSUE THREAD

**Issue by jgillham** [dviramontes/impress#11]
The mock MongoDB will be a JS based DB that will be have the same API as MongoDB but limited scalability. The mock DB will be used to start the project out and provide a testing framework. The project should be designed so that switching to the real MongoDB later will be trivial.

I believe there is an existing mock DB available.
- [ ] 1. Find an existing mock DB.
  - [c] Try https://github.com/openhoat/mongodb-fs
  - [c] Try https://github.com/openhoat/monckoose
  - [c] Try https://github.com/mbostock/mongo-mock
  - [c] Try https://github.com/ubicast/MockMockWeb
- [ ] 2. Decide on the best alternative.
- [ ] 3. Fork that project.
- [ ] 4. Refactor the fork to make it work with the project.
- [ ] 5. Set our project up so that it can readily import this dependency.

ADDITIONAL RESOURCES:

Fixtures:
- https://github.com/tdegrunt/mongodb-fixtures
- https://github.com/powmedia/pow-mongodb-fixtures
- https://github.com/powmedia/mongoose-fixtures

Test helpers and Fixtures:
- http://mongodb.github.io/node-mongodb-native/github/github.html#test-helpers-and-libraries

**Comment by jgillham** [comment]
After trying Mongodb-fs and monckoose, here are my conclusions.
- Both projects are immature.
- Mongodb-fs failed to work with Mongoose on our model for the insert operation.
- Monckoose has not implemented most operations.

I conclude that some additional work is required to develop one of these projects.

**Comment by dviramontes** [comment]
May I also recommend that we look at leveldb from google ?

**Comment by jgillham** [comment]
Looks cool! I noticed that it is written in c. Will that play well with AWS?

**Comment by jgillham** [comment]
After trying mongo-mock, here are my conclusions:
- Very immature.

I conclude that a significant amount of developement would have to be invested in this project.

**Comment by dviramontes** [comment]
yeah that should be fine, we would have to install it and configure it ourselves, that the only difference. But its a simple programming interface, very intuitive.

There are of course, drivers for node.js checkout this talk:

https://vimeo.com/70232116

also @jnels124 if you have time.
Very interesting.

**Comment by jgillham** [comment]
Looks very much worth my time!

**Comment by jgillham** [comment]
Here is npm link:
https://npmjs.org/package/leveldb
